### PR TITLE
Add docs: Deploy deployment into a specified set of target clusters

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind documentation

**What this PR does / why we need it**:

Add docs: Deploy deployment into a specified set of target clusters.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Hi @Tingtal, can you help review it?

File: docs/userguide/resource-propagating.md [hunk 1]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
 
 ## Deploy deployment into a specified set of target clusters
 
+`.spec.placement.clusterAffinity` field of PropagationPolicy represents scheduling restrictions to a certain set of clusters. If not set, any cluster can be scheduling candidate.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:144]:
```suggestion
`.spec.placement.clusterAffinity` field of PropagationPolicy represents scheduling restrictions on a certain set of clusters, without which any cluster can be scheduling candidates.
```

File: docs/userguide/resource-propagating.md [hunk 2]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+
+It has four fields to set:
+- LabelSelector
+- FieldSelector
+- ClusterNames
+- ExcludeClusters
+
+### LabelSelector
+
+LabelSelector is a filter to select member clusters by labels, it uses `*metav1.LabelSelector` type. If non-nil and non-empty, only the clusters match this filter will be selected.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:154]:
```suggestion
LabelSelector is a filter to select member clusters by labels. It uses `*metav1.LabelSelector` type. If it is non-nil and non-empty, only the clusters match this filter will be selected.
```

File: docs/userguide/resource-propagating.md [hunk 3]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+          values:
+          - us
+    ...
+```
+
+For a description of `matchLabels` and `matchExpressions`, you can refer to [Resources that support set-based requirements](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements).
+
+### FieldSelector
+
+FieldSelector is a filter to select member clusters by fields. If non-nil and non-empty, only the clusters match this filter will be selected.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:197]:
```suggestion
FieldSelector is a filter to select member clusters by fields. If it is non-nil and non-empty, only the clusters match this filter will be selected.
```

File: docs/userguide/resource-propagating.md [hunk 4]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+        - key: region
+          operator: NotIn
+          values:
+          - cn-south-1
+    ...
+```
+
+If multiple `matchExpressions` are specified in the `fieldSelector`, the cluster must match all `matchExpressions`.
+
+The `key` value in `matchExpressions` field now support three values: `provider`, `region`, `zone`, which correspond to the `.spec.provider`, `.spec.region`, `.spec.zone` fields of the Cluster object respectively.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:225]:
```suggestion
The `key` in `matchExpressions` now supports `provider`, `region`, and `zone`, which correspond to the `.spec.provider`, `.spec.region`, and `.spec.zone` fields of the Cluster object, respectively.
```

File: docs/userguide/resource-propagating.md [hunk 5]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+
+The `operator` value in `matchExpressions` field now support two operator: `In` and `NotIn`.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:227]:
```suggestion
The `operator` in `matchExpressions` now supports `In` and `NotIn`.
```

File: docs/userguide/resource-propagating.md [hunk 6]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+
+### ClusterNames
+
+User can set the ClusterNames field to specify the name of the selected cluster list.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:231]:
```suggestion
Users can set the `ClusterNames` field to specify the selected clusters.
```

File: docs/userguide/resource-propagating.md [hunk 7]
```diff
@@ -141,6 +141,132 @@ kubectl delete deployment nginx
...
+    clusterAffinity:
+      clusterNames:
+        - member1
+        - member2
+    ...
+```
+
+### ExcludeClusters
+
+User can set the ExcludeClusters fields to specify the cluster name list, indicating that these clusters are ignored.
```
Comment (Tingtal) [inline docs/userguide/resource-propagating.md:252]:
```suggestion
User can set the `ExcludeClusters` fields to specify the clusters to be ignored.
```

**Review by Tingtal — CHANGES_REQUESTED** [review]

**Comment by XiShanYongYe-Chang** [comment]
@Tingtal, thanks for your review.

**Review by RainbowMango — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**